apps/wallet/src/ui/app/pages/swap/index.tsx

```diff
@@ -0,0 +1,317 @@
+
+	const formattedTokenBalance = formattedBalance.replace(/,/g, '');
+	const symbol = coinMetadata.data?.symbol ?? '';
+
+	const coinDecimals = coinMetadata.data?.decimals ?? 0;
+	const maxBalanceInMist = rawBalance || '0';
+
+	const validationSchema = useMemo(() => {
+		return z.object({
+			amount: z.string().transform((value, context) => {
+				const bigNumberValue = new BigNumber(value);
+
+				if (!value.length) {
+					context.addIssue({
+						code: 'custom',
+						message: 'Amount is required.',
+					});
+					return z.NEVER;
+				}
+
+				if (bigNumberValue.lt(0)) {
+					context.addIssue({
+						code: 'custom',
+						message: 'Amount must be greater than 0.',
+					});
+					return z.NEVER;
+				}
+
+				if (bigNumberValue.shiftedBy(coinDecimals).gt(BigInt(maxBalanceInMist).toString())) {
+					context.addIssue({
+						code: 'custom',
+						message: 'Not available in account',
+					});
+					return z.NEVER;
+				}
+
+				return value;
+			}),
+			isPayAll: z.boolean(),
+			quoteAssetType: z.string(),
+			allowedMaxSlippagePercentage: z.string().transform((percent, context) => {
+				const numberPercent = Number(percent);
+
+				if (numberPercent < 0 || numberPercent > 100) {
+					context.addIssue({
+						code: 'custom',
+						message: 'Value must be between 0 and 100.',
+					});
+					return z.NEVER;
+				}
+
+				return percent;
+			}),
+		});
+	}, [maxBalanceInMist, coinDecimals]);
+
+	const form = useZodForm({
+		mode: 'all',
+		schema: validationSchema,
+		defaultValues: initialValues,
+	});
+
+	const {
+		register,
+		getValues,
+		setValue,
+		control,
+		handleSubmit,
+		trigger,
+		formState: { isValid, isSubmitting, errors },
+	} = form;
+
+	const renderButtonToCoinsList = useMemo(() => {
+		return (
+			recognized.length > 1 &&
+			recognized.some((coin) => allowedSwapCoinsList.includes(coin.coinType))
+		);
+	}, [recognized]);
+
+	const isPayAll = getValues('isPayAll');
+	const amount = useWatch({
+		name: 'amount',
+		control,
+	});
+	const quoteAssetType = getValues('quoteAssetType');
+	const amountAsBigInt = new BigNumber(amount);
+	const { rawValue } = useSuiBalanceInUSDC(amountAsBigInt);
+	const atcText = useMemo(() => {
+		return getSwapPageAtcText(symbol, quoteAssetType, coinsMap);
+	}, [symbol, quoteAssetType, coinsMap]);
+
+	const suiBalanceInMIST = new BigNumber(amount || '0').shiftedBy(SUI_DECIMALS).toString();
+
+	const { data: currentEstimatedData } = useGetEstimateSuiToUSDC({
+		balanceInMist: suiBalanceInMIST,
+		signer,
+		accountCapId,
+		coinType: activeCoinType || '',
+	});
+
+	const recognizedPackagesList = useRecognizedPackages();
+
+	const txnSummary = useTransactionSummary({
+		transaction: currentEstimatedData as DryRunTransactionBlockResponse,
+		recognizedPackagesList,
+		currentAddress: activeAccountAddress,
+	});
+
+	const totalGas = txnSummary?.gas?.totalGas;
+
+	const { data } = useCoinsByBalance({ coinType: activeCoinType!, balance: suiBalanceInMIST });
```

Comment:
This is only used in the market order mutation, so you don't need to fetch this in the hook. Just make a utility function you can call instead.

Reply:
This also is used in `useGetEstimateSuiToUSDC`